src/window.rs

```diff
@@ -291,7 +293,59 @@ impl Window {
                     }
                 };
 
-                // handle Windows and Linux non-decoration window resize
+                /* handle context menu */
+                // TODO(context-menu): should create on ShowContextMenu event
+
+                #[cfg(linux)]
+                {
+                    let is_click_on_context_menu =
+                        self.is_position_on_context_menu(compositor, position);
+
+                    if !is_click_on_context_menu {
+                        if *state == ElementState::Pressed {
+                            match *button {
+                                winit::event::MouseButton::Left => {
+                                    if self.close_context_menu(sender) {
+                                        // return here to bypass following mouse event for underlying element
+                                        return;
+                                    }
+                                }
+                                winit::event::MouseButton::Right => {
+                                    // Close old context menu
+                                    self.close_context_menu(sender);
+                                    // Create new context menu
+                                    self.context_menu = Some(self.show_context_menu(sender));
+                                    return;
+                                }
+                                _ => {}
+                            }
+                        } else if *state == ElementState::Released {
+                            match *button {
+                                winit::event::MouseButton::Right => {
+                                    if self.context_menu.is_some() {
+                                        return;
+                                    }
+                                }
+                                _ => {}
+                            }
+                        }
+                    }
+                    // TODO(context-menu): ignore first release event after context menu open or close to prevent click on backgound element
```

Comment:
```suggestion
                    // TODO(context-menu): ignore first release event after context menu open or close to prevent click on background element
```